scarlet2/frame.py

```diff
@@ -38,75 +40,92 @@ def pixel_size(self):
             Coordinates in the pixel space
-        """
-        pix = jnp.asarray(pixel, dtype=np.float64).reshape(-1, 2)
 
-        if self.wcs is not None:
-            wcs = self.wcs.celestial  # only use celestial portion
-            # x/y instead of y/x:
-            pix = jnp.flip(pix, axis=-1)
-            sky = jnp.array(wcs.pixel_to_world_values(pix))
-        else:
-            sky = pix
+        Returns
+        ----------
+        astropy.coordinates.SkyCoord
+        """
+        pixels = pixels.reshape(-1, 2)
 
-        if sky.size == 2:
-            return sky[0]
-        return sky
+        assert self.wcs is not None
+        wcs = self.wcs.celestial # only use celestial portion
+        sky_coord = [SkyCoord.from_pixel(pixel[0], pixel[1], wcs) for pixel in pixels]
 
+        return sky_coord
+    
     def convert_pixel_to(self, target, pixel=None):
         """Converts pixel coordinates from this frame to `target` Frame
 
-        Parameters
-        ----------
-        target: `~scarlet.Frame`
-            target frame
-        pixel: `tuple` or list ((y1,y2, ...), (x1, x2, ...))
-            pixel coordinates in this frame
-            If not set, convert all pixels in this frame
-
-        Returns
-        -------
-        coord_target: `tuple`
-            coordinates at the location of `coord` in the target frame
+            Parameters
+            ----------
+            target: `~scarlet2.Frame`
+                target frame
+            pixel: `array`, pixel coordinates in this frame
+                If not set, convert all pixels in this frame
+
+            Returns
+            -------
+            coord_target: `array`
+                coordinates at the location of `coord` in the target frame
         """
+
         if pixel is None:
-            y, x = np.indices(self.bbox.shape[-2:], dtype=np.float64)
-            pixel = np.stack((y.flatten(), x.flatten()), axis=1)
+            y, x = jnp.indices(self.bbox.shape[-2:], dtype="float32")
+            pixel = jnp.stack((y.flatten(), x.flatten()), axis=1)
 
         ra_dec = self.get_sky_coord(pixel)
-        pixel_ = target.get_pixel(ra_dec)
-
-        if pixel_.size == 2:  # only one coordinate pair
-            return pixel_[0]
-        return pixel_
+        return target.get_pixel(ra_dec)
+    
+    def u_to_pixel(self, size):
+        """Converts a size un astropy.units.Quantity to pixel size according 
+           to this frame WCS
+
+            Parameters
+            ----------
+            size: `astropy.units.Quantity`, must be PhysicalType("angle")
+
+            Returns
+            -------
+            size in pixels
+        """
+        assert u.get_physical_type(size) == "angle"
+        # first computer the pixel size
+        pixel_size = get_pixel_size(
+            get_affine(self.wcs.celestial) # only use celestial portion
+        ) # in deg/pixel
+        
+        return size.to(u.deg).value / pixel_size
```

Comment:
I know it's arbitrary, but degrees is an inconvenient unit for pixel sizes. I'd choose arcsec instead if this doesn't cause any trouble elsewhere